packages/build-tools/src/context.ts

```diff
@@ -282,6 +282,11 @@ export class BuildContext<TJob extends Job> {
   }
 
   public getReactNativeProjectDirectory(baseDirectory = this.buildDirectory): string {
+    if (!this.job.platform) {
+      // Generic jobs start from base directory.
+      return baseDirectory;
+    }
```

Comment:
I know it's all tricky, but do we even need to use BuildContext for generic jobs? I mean it seems like a build job only concept.

Reply:
Hmm, so I think a lot of `BuildContext` is reusable for generic jobs too. All build phases logic, env, artifacts…

I agree some parts are build-specific, like `getReactNativeProjectDirectory`, `appConfig`.

Maybe what we'll need to do is have a `TaskContext<TJob>` which would have build phases, env, artifacts and have a `BuildContext < TaskContext<BuildJob>` with `appConfig` and stuff. I didn't want to do too much in one pull request though, so my assumption for now is that `BuildContext` is like `TaskContext`.